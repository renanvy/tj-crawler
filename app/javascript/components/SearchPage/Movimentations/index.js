import React from "react";
import moment from "moment";
import PropTypes from "prop-types";

export default function Movimentations(props) {
  const { movimentations } = props;

  return (
    <div className="Movimentations">
      <div className="col s12 m8">
        <ul className="collection with-header">
          <li className="collection-header">
            <h1 className="Movimentations_title">Movimentações</h1>
          </li>

          {movimentations.map(movimentation => (
            <li
              key={`movimentation-${movimentation.id}`}
              className="collection-item"
            >
              <span className="Movimentation_date">
                {moment(movimentation.date).format("DD/MM/YYYY")}
              </span>

              <p className="Movimentation_description">
                {movimentation.description}
              </p>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
}

Movimentations.propTypes = {
  movimentations: PropTypes.array.isRequired
};
