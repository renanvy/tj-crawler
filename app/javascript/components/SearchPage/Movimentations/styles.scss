.Movimentations {
  margin-top: 20px;

  &_title {
    margin: 0;
    padding: 0;
    font-size: 18px;
    font-weight: 500;
  }
}

.Movimentation {
  &_date {
    display: flex;
    font-size: 12px;
    color: #7c7c7c;
  }

  &_description {
    margin-top: 10px;
    font-size: 14px;
  }
}