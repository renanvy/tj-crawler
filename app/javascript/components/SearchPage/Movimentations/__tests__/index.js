import React from "react";
import renderer from "react-test-renderer";
import Movimentations from "../index";

describe("Movimentations", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<Movimentations movimentations={[]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
