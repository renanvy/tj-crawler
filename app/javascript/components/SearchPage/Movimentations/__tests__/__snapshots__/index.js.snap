// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movimentations snapshot renders 1`] = `
<div
  className="Movimentations"
>
  <div
    className="col s12 m8"
  >
    <ul
      className="collection with-header"
    >
      <li
        className="collection-header"
      >
        <h1
          className="Movimentations_title"
        >
          Movimentações
        </h1>
      </li>
    </ul>
  </div>
</div>
`;
